### ISSUE TITLE
Incompatibility with oe_theme, oe_oembed and ckeditor5

### ISSUE THREAD

**Issue by TechnoTim2010** [openeuropa/oe_theme#1392]
Hi
We recently deployed Drupal 10.2.2 on our site https://digital-strategy.ec.europa.eu/
We need to, by necessity, because ckeditor version 4 is EOL and unsupported, migrate our text format editors to use ckeditor5 rather than ckeditor4.
Whilst work is ongoing to rewrite custom plugins for ckeditor4 to work with ckeditor5  we wanted the media embed plugin provided as an open europa component to work in the meantime.
However we found several issues:
The first was a warning on D10.2.2 whilst still using ckeditor4 on status report for Drupal 10 or Embed button Ckeditor5 compatibility complaining about the embed icon being a png See attachment.
![ckeditor media plugin png error](https://github.com/openeuropa/oe_theme/assets/65389784/74ba930b-bde4-44b3-93a6-8eee6aa4068a)
So I checked and the version of oe_embed component installed by composer for oe_theme version 3.9.0 is 0.8.0. Checking the open europa repository, the version we required of oe_embed is 0.9.0 which aside from being built with ckeditor5 in mind has the icon as an svg file rather than a png file.
Ckeditor5 seems to only accept .svg files as plugin icons. This we accept and undertstand.
However the svg default file is very different from the version we wanted and is familiar to our users, but there is no way to upload a replacement. The error indicates that the user uploading the file has insufficient permissions. The upload location appears to be public:/embed_buttons or sites/default/files/embed_buttons but no amount of playing with file and folder permissions allowed replacement of the default icon with the one we want to use.

In summary there are two issues:
1) Oe-theme version 3.9.0 which installs version 0.8.0 of oe_oembed currently prevents migration to ckeditor5 with use of the media embed plugin. This requires a separate install using composer require of openeuropa/oe_oembed:^0.9.0
2) The new version does not allow replacement of the default icon embed.svg with any user required file. (this includes trying it with a copy of the supplied embed.svg. 

We have a working solution for now, but you might want to forewarn other Drupal development teams of this issue or perhaps have a working solution that works with ckeditor5.

Regards

Tim

**Comment by brummbar** [comment]
Hello @TechnoTim2010 . Regarding your points:

1. oe_theme doesn't have oe_oembed in its composer.json, so it's not a dependency. The lock file belongs to the project that requires oe_theme/oe_oembed.
2. I did a clean install of oe_oembed and I was able to upload the icon without issue. The file upload is not handled by oe_oembed, it's core file field. Insufficient permissions are most likely caused by the folders in your installation.
